### ISSUE TITLE
Enhanced MIA

### ISSUE THREAD

**Issue by Ty0ng** [privacytrustlab/ml_privacy_meter#99]
Hi what happened to the codes/folder for Enhanced MIA?

**Comment by rzshokri** [comment]
With this new version of privacy meter you can reproduce many of the results in enhanced MIA paper, and other papers. We will soon be adding more info plus access to the older paper code.

**Comment by Ty0ng** [comment]
Okay, thanks. Does this new version support python version >=3.6? However, I got this error on Colab when following the instruction for installation: Package 'privacy-meter' requires a different Python: 3.8.10 not in '>=3.9.0'

**Comment by changhongyan123** [comment]
Hi @Ty0ng,

I wanted to follow up on the issues you reported. We have provided the pointer to the previous Enhanced MIA implementation in the [research/readme.md](https://github.com/privacytrustlab/ml_privacy_meter/tree/master/research) file. Please refer to that for further details.

Regarding the issues on Colab, we have included a workaround in the [tutorial/readme.md](https://github.com/privacytrustlab/ml_privacy_meter/tree/master/tutorials) file. Could you please check if the provided solution resolves your problem?

If you have any further questions or concerns, please let us know.

**Comment by Ty0ng** [comment]
Yes the code worked. Thank you!

I have another question, for the reference attack. According to the paper, the target dataset used to train the target model should be different from the reference dataset for the reference model. What happens if the reference dataset is a subset of the target dataset? 
For example, training the target model on the full Cifar10 dataset and the reference models on subset of the Cifar10 (5000 points).

**Comment by changhongyan123** [comment]
Hi @Ty0ng, 

If the reference dataset is a subset of the target dataset, it is possible that the reference model may not accurately simulate the target model. The high-level goal of the reference attack is to estimate the loss distribution of the target model when the target point is not present in the training dataset. To achieve this, it is essential that the reference model and the target model have similar performance levels.

For example, let us consider a scenario where the target model is trained on the entire CIFAR10 dataset (50K), but the reference models are only trained on a smaller subset of the dataset (10K). In such a case, it is likely that most of the target data points will have high loss values from the reference models. However, it is important to note that this does not necessarily mean that all target points will have high loss values when they are not in the training dataset. Rather, this result may be due to the poor simulation of the reference model.

On the other hand, existing research also considers situations where the reference models' training dataset overlaps with the training dataset of the target model. In such cases, the reference attack can still be validated. Specifically, in the following paper, the authors construct the loss distribution for the target point when it is used for the reference model training versus when it is not used for the reference model training ([Membership Inference Attacks From First Principles
](https://arxiv.org/abs/2112.03570)). You can find more details in the paper. 

If you have any further questions or concerns, please let us know.

**Comment by changhongyan123** [comment]
Hi @Ty0ng ,

To evaluate the privacy risk of a machine learning model, it's important to understand the security game being played, as outlined in Section 3.1 of in the [paper](https://arxiv.org/abs/2111.09679). The privacy loss of the target model with respect to its training dataset is the adversary's success in winning the security game over multiple repetitions. The attack error depends on various factors listed in Section 3.2 of the paper. 

Regarding your question, if the reference dataset is a subset of the target dataset, you are providing the adversary with additional information and changing the security game. Specifically, in this scenario, the adversary's objective would be to infer membership information about the target point, given the knowledge about a subset of the target model's training dataset. The results of the attack would have a different meaning compared to the reference attack evaluated in the paper. For a more thorough discussion on this topic, please refer to Section 4 of the paper.

I hope this explanation helps.
